search/collector/eligible.go

```diff
@@ -55,8 +51,10 @@ func makeEligibleDocumentMatchHandler(ctx *search.SearchContext) (search.Documen
 				return nil
 			}
 
-			// No elements removed from the store here.
-			_ = ec.store.Add(d)
+			copyOfID := make([]byte, len(d.IndexInternalID))
+			copy(copyOfID, d.IndexInternalID)
+			ec.ids = append(ec.ids, copyOfID)
+			ctx.DocumentMatchPool.Put(d)
```

Comment:
@metonymic-smokey real quick why is this `DocumentMatchPool.Put(..)` necessary?

Reply:
This is to avoid hitting into the DocMatchPoolTooSmall issue - by putting it back in the pool, the pool will always have a doc match and won't have to allocate more doc matches.

Reply:
Cheers.